pkg/northbound/admin/configuration.go

```
@@ -0,0 +1,84 @@
+)
+
+// GetConfiguration returns response with the requested configuration
+func (s Server) GetConfiguration(ctx context.Context, req *admin.GetConfigurationRequest) (*admin.GetConfigurationResponse, error) {
+	log.Infof("Received GetConfiguration request: %+v", req)
+	conf, err := s.configurationsStore.Get(ctx, req.TargetID)
+	if err != nil {
+		log.Warnf("GetConfiguration %+v failed: %v", req, err)
+		return nil, errors.Status(err).Err()
+	}
+	return &admin.GetConfigurationResponse{Configuration: conf}, nil
+}
+
+// ListConfigurations provides stream listing all configurations
+func (s Server) ListConfigurations(req *admin.ListConfigurationsRequest, stream admin.ConfigurationService_ListConfigurationsServer) error {
+	log.Infof("Received ListConfigurations request: %+v", req)
+	configurations, err := s.configurationsStore.List(stream.Context())
+	if err != nil {
+		log.Warnf("ListConfigurations %+v failed: %v", req, err)
+		return errors.Status(err).Err()
+	}
+	for _, conf := range configurations {
+		err := stream.Send(&admin.ListConfigurationsResponse{Configuration: conf})
+		if err != nil {
+			log.Warnf("ListConfigurations %+v failed: %v", req, err)
+			return errors.Status(err).Err()
+		}
+	}
+	return nil
+}
+
+// WatchConfigurations provides stream with events representing configuration changes
+func (s Server) WatchConfigurations(req *admin.WatchConfigurationsRequest, stream admin.ConfigurationService_WatchConfigurationsServer) error {
+	log.Infof("Received WatchConfigurations request: %+v", req)
+	var watchOpts []configuration.WatchOption
+	if !req.Noreplay {
+		watchOpts = append(watchOpts, configuration.WithReplay())
+	}
+
+	ch := make(chan configapi.ConfigurationEvent)
+	if err := s.configurationsStore.Watch(stream.Context(), ch, watchOpts...); err != nil {
+		log.Warnf("WatchConfigurationsRequest %+v failed: %v", req, err)
+		return errors.Status(err).Err()
+	}
+
+	return s.streamConfigurations(stream, ch)
```

I think we should return the error here and convert it to status before returning it back.

---

Fixed